### PR TITLE
Add table aws_ecs_task_definition. Closes #171

### LINKED ISSUE #171: Add table aws_ecs_task_definition
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ecs

### PR DESCRIPTION
**Pull request by khushboo9024**
```No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_ecs_task_definition []

PRETEST: tests/aws_ecs_task_definition

TEST: tests/aws_ecs_task_definition
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ecs_task_definition.named_test_resource: Creating...
aws_ecs_task_definition.named_test_resource: Creation complete after 3s [id=turbottest69950]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = arn:aws:ecs:us-east-1:013122550996:task-definition/turbottest69950:1
resource_id = turbottest69950
resource_name = turbottest69950

Running SQL query: test-get-query.sql
[
  {
    "cpu": "<null>",
    "family": "turbottest69950",
    "memory": "<null>",
    "status": "ACTIVE",
    "task_definition_arn": "arn:aws:ecs:us-east-1:013122550996:task-definition/turbottest69950:1"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "cpu": "<null>",
    "family": "turbottest69950",
    "memory": "<null>",
    "status": "ACTIVE",
    "task_definition_arn": "arn:aws:ecs:us-east-1:013122550996:task-definition/turbottest69950:1"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:ecs:us-east-1:013122550996:task-definition/turbottest69950:1"
    ],
    "tags": {
      "name": "turbottest69950"
    },
    "task_definition_arn": "arn:aws:ecs:us-east-1:013122550996:task-definition/turbottest69950:1",
    "title": "turbottest69950:1"
  }
]
✔ PASSED

POSTTEST: tests/aws_ecs_task_definition

TEARDOWN: tests/aws_ecs_task_definition

SUMMARY:

1/1 passed.
```

```### Basic info

+----------------------------------------------------------------+--------+--------------+-------------+--------+----------------------------------+
| task_definition_arn                                            | cpu    | network_mode | title       | status | tags                             |
+----------------------------------------------------------------+--------+--------------+-------------+--------+----------------------------------+
| arn:aws:ecs:us-east-2:013122550996:task-definition/testinggg:1 | <null> | awsvpc       | testinggg:1 | ACTIVE | {}                               |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test-89:1   | <null> | bridge       | test-89:1   | ACTIVE | {}                               |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test:1      | <null> | bridge       | test:1      | ACTIVE | {}                               |
| arn:aws:ecs:us-east-1:013122550996:task-definition/test09:1    | 1024   | host         | test09:1    | ACTIVE | {"test09":"check","tom":"jerry"} |
| arn:aws:ecs:us-east-1:013122550996:task-definition/test01:1    | 1024   | host         | test01:1    | ACTIVE | {"FOO":"BAR","FOO1":"BAR1"}      |
+----------------------------------------------------------------+--------+--------------+-------------+--------+----------------------------------+

### Number of containers attached to task definitions

+----------------------------------------------------------------+------------------+
| task_definition_arn                                            | no_of_conatiners |
+----------------------------------------------------------------+------------------+
| arn:aws:ecs:us-east-1:013122550996:task-definition/test01:1    | 1                |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test-89:1   | 2                |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test-89:1   | 2                |
| arn:aws:ecs:us-east-1:013122550996:task-definition/test09:1    | 1                |
| arn:aws:ecs:us-east-2:013122550996:task-definition/testinggg:1 | 1                |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test:1      | 1                |
+----------------------------------------------------------------+------------------+

### List of containers with elevated privileges on the host container instance

+----------------------------------------------------------------+------------+----------------+
| task_definition_arn                                            | privileged | container_name |
+----------------------------------------------------------------+------------+----------------+
| arn:aws:ecs:us-east-2:013122550996:task-definition/testinggg:1 | true       | Elasti-592     |
+----------------------------------------------------------------+------------+----------------+

```
```
### List task definitions with containers where logging is disabled

```sql
select
  task_definition_arn,
  cd ->> 'Name' as container_name,
  cd ->> 'LogConfiguration' as log_configuration
from
  aws_ecs_task_definition,
  jsonb_array_elements(container_definitions) as cd
where
 cd ->> 'LogConfiguration' is null;
 +----------------------------------------------------------------+----------------+-------------------+
| task_definition_arn                                            | container_name | log_configuration |
+----------------------------------------------------------------+----------------+-------------------+
| arn:aws:ecs:us-east-2:013122550996:task-definition/test-89:1   | ubuntu         | <null>            |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test-89:1   | testing        | <null>            |
| arn:aws:ecs:us-east-2:013122550996:task-definition/test:1      | ubuntu         | <null>            |
| arn:aws:ecs:us-east-1:013122550996:task-definition/test09:1    | test05         | <null>            |
| arn:aws:ecs:us-east-2:013122550996:task-definition/testinggg:1 | Elasti-592     | <null>            |
| arn:aws:ecs:us-east-1:013122550996:task-definition/test01:1    | test05         | <null>            |
+----------------------------------------------------------------+----------------+-------------------+
```

### CODE REVIEW

File: aws/table_aws_ecs_task_definition.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+			{
+				Name:        "volumes",
+				Description: "The list of volume definitions for the task.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getEcsTaskDefinition,
+				Transform:   transform.FromField("TaskDefinition.Volumes"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags associated with target group.",
```
Comment (Subhajit97) [inline aws/table_aws_ecs_task_definition.go:179]:
```suggestion
				Description: "A list of tags associated with task.",
```

File: aws/table_aws_ecs_task_definition.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	// Create Session
+	svc, err := EcsService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+	params := &ecs.ListTaskDefinitionsInput{}
+	pagesLeft := true
+
+	for pagesLeft {
+		result, err := svc.ListTaskDefinitions(params)
```
Comment (Subhajit97) [inline aws/table_aws_ecs_task_definition.go:229]: Can't we use `ListTaskDefinitionsPages` for pagination?
https://github.com/aws/aws-sdk-go/blob/67f532fc99ebc262a50fa05500b3462065b75569/service/ecs/api.go#L3317

File: aws/table_aws_ecs_task_definition.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+//// TRANSFORM FUNCTIONS
+
+func getAwsEcsTaskDefinitionTurbotData(ctx context.Context, d *transform.TransformData) (interface{},
+	error) {
+	param := d.Param.(string)
+	ecsTaskDefinition := d.HydrateItem.(*ecs.DescribeTaskDefinitionOutput)
+
+	// Get resource title
+	arn := ecsTaskDefinition.TaskDefinition.TaskDefinitionArn
+	title := strings.Split(*arn, "/")[len(strings.Split(*arn, "/"))-1]
```
Comment (Subhajit97) [inline aws/table_aws_ecs_task_definition.go:301]:
```suggestion
	splitArn := strings.Split(*arn, "/")
	title := splitArn[len(splitArn)-1]
```

File: docs/tables/aws_ecs_task_definition.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+### Number of containers attached to task definitions
+
+```sql
+select
+  task_definition_arn,
+  jsonb_array_length(container_definitions) as no_of_conatiners
+from
+  aws_ecs_task_definition,
+  jsonb_array_elements(container_definitions) as cd;
```
Comment (Subhajit97) [inline docs/tables/aws_ecs_task_definition.md:29]: Do we need this?

File: aws-test/tests/aws_ecs_task_definition/test-get-query.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+select task_definition_arn, cpu, family, memory, status
```
Comment (Subhajit97) [inline aws-test/tests/aws_ecs_task_definition/test-get-query.sql:1]: Also validate `account_id`, `region` and `partition`.

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes

**Review by Subhajit97 — APPROVED** [review]

File: docs/tables/aws_ecs_task_definition.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+```sql
+select
+  task_definition_arn,
+  cd ->> 'Privileged' as privileged,
+  cd ->> 'Name' as container_name
+from
+  aws_ecs_task_definition,
+  jsonb_array_elements(container_definitions) as cd
+  where cd ->> 'Privileged' = 'true' ;
```
Comment (Subhajit97) [inline docs/tables/aws_ecs_task_definition.md:42]:
```suggestion
where 
  cd ->> 'Privileged' = 'true';
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@khushboo9024 There is a small change in docs. Please make the changes.

**Review by Subhajit97 — APPROVED** [review]

File: aws/table_aws_ecs_task_definition.go [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+	param := d.Param.(string)
+	ecsTaskDefinition := d.HydrateItem.(*ecs.DescribeTaskDefinitionOutput)
+
+	// Get resource title
+	arn := ecsTaskDefinition.TaskDefinition.TaskDefinitionArn
+	splitArn := strings.Split(*arn, "/")
+	title := splitArn[len(splitArn)-1]
+
+	if param == "Tags" {
+		// Get the resource tags
```
Comment (Subhajit97) [inline aws/table_aws_ecs_task_definition.go:296]:
```suggestion
		if ecsTaskDefinition.Tags == nil {
		  return nil, nil
		}
		
		// Get the resource tags
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
There is a minor change required. Please see the comment.

**Review by Subhajit97 — APPROVED** [review]
LGTM

File: docs/tables/aws_ecs_task_definition.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# Table: aws_ecs_task_definition
+
+A task definition is required to run Docker containers in Amazon ECS.
+
+## Examples
+
+### Basic info
```
Comment (rajmohanty17) [inline docs/tables/aws_ecs_task_definition.md:7]: Would recommend to add  network_mode to the query to check what's the  ECS task networking type.

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Added to include meaningful col.

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Pull request merged**